spacelift/resource_api_key.go

```diff
@@ -0,0 +1,481 @@
+	var targetRoleName string
+	switch accessLevel {
+	case "READ":
+		targetRoleName = "Space reader"
+	case "WRITE":
+		targetRoleName = "Space writer"
+	case "ADMIN":
+		targetRoleName = "Space admin"
+	default:
+		// For anything else, treat it as a custom role name and search for it directly
+		targetRoleName = accessLevel
+	}
+
+	// Find the role ID for the target role name (built-in or custom)
+	for _, role := range query.Roles {
+		if role.Name == targetRoleName {
+			return role.ID, nil
+		}
+	}
+
+	return "", fmt.Errorf("role not found: %s", accessLevel)
+}
+
+// manageAPIKeyRoleBindings handles creating/updating role bindings for an API key
+func manageAPIKeyRoleBindings(ctx context.Context, client *internal.Client, apiKeyID string, accessRules []interface{}) error {
+	if len(accessRules) == 0 {
+		return nil // No access rules to manage
+	}
+
+	var bindings []structs.ApiKeyRoleBindingInput
+	for _, rule := range accessRules {
+		ruleMap := rule.(map[string]interface{})
+		spaceID := ruleMap["space_id"].(string)
+		accessLevel := ruleMap["role"].(string)
+
+		roleID, err := getRoleIDForAccessLevel(ctx, client, accessLevel)
+		if err != nil {
+			return fmt.Errorf("could not get role ID for access level %s: %v", accessLevel, err)
+		}
+
+		bindings = append(bindings, structs.ApiKeyRoleBindingInput{
+			APIKeyID: graphql.ID(apiKeyID),
+			RoleID:   graphql.ID(roleID),
+			SpaceID:  graphql.ID(spaceID),
+		})
+	}
+
+	// Create role bindings using batch mutation
+	var mutation struct {
+		RoleBindings []structs.APIKeyRoleBinding `graphql:"apiKeyRoleBindingBatchCreate(input: $input)"`
+	}
+
+	input := structs.ApiKeyRoleBindingBatchInput{
+		Bindings: bindings,
+	}
+
+	variables := map[string]interface{}{
+		"input": input,
+	}
+
+	if err := client.Mutate(ctx, "APIKeyRoleBindingBatchCreate", &mutation, variables); err != nil {
+		return fmt.Errorf("could not create role bindings: %v", err)
+	}
+
+	return nil
+}
+
+func resourceAPIKeyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*internal.Client)
+
+	// Create the API key first
+	var mutation struct {
+		APIKey structs.APIKey `graphql:"apiKeyCreate(input: $input)"`
+	}
+
+	variables := map[string]interface{}{
+		"input": apiKeyCreateInput(d),
+	}
+
+	if err := client.Mutate(ctx, "APIKeyCreate", &mutation, variables); err != nil {
+		return diag.Errorf("could not create API key %v: %v", d.Get("name").(string), err)
+	}
+
+	apiKeyID := mutation.APIKey.ID
+	d.SetId(apiKeyID)
+
+	// Capture the secret immediately - it's only available once during creation
+	if mutation.APIKey.Secret != "" {
+		d.Set("secret", mutation.APIKey.Secret)
+	}
+
+	// Also set other fields that are available from the creation response
+	d.Set("name", mutation.APIKey.Name)
+	d.Set("type", string(mutation.APIKey.Type))
+
+	// Handle access rules using role bindings
+	if accessRulesSet, ok := d.Get("access_rule").(*schema.Set); ok && accessRulesSet.Len() > 0 {
```

Comment:
I believe this part is already handled by the backend when creating the API key.